docs/GLOSSARY.md

```diff
@@ -1,122 +1,180 @@
 # Glossary
 
-Index of terminology used in Randovania. Certain terms sometimes are used only in the GUI, with more technical ones being used in code or CLI.
+Index of terminology used in Randovania. Some terms are used only in the GUI, with more technical ones being used in code or CLI.
+
+## Action
+
+Anything the [Generator](#generator) or [Solver](#solver) can do which alters the game's state. Actions include obtaining [Resources](#resouce), crossing [Connections](#connection) between [Nodes](#node), and using [Docks](#dock) or [Teleporters](#teleporter).
 
 ## Area
 
-TODO
+A small chunk of a game's playable space, often only a single room. May contain one or more [Nodes](#node).
+
+Examples:
+- Phazon Mining Tunnel in Metroid Prime
+- Big Pink in Super Metroid
+
+## Burger King / BK
+
+A strictly informal term used to describe the situation of being unable to progress in a [Multiworld](#multiworld) [Session](#session). This occurs when a player has exhausted all readily available [Checks](#item-check) and must wait for another player to send them [Progression](#progression) before they can continue. The term was coined after a player in this state left to get food from the eponymous restaurant and had yet to obtain Progression by the time they returned.
 
 ## Configuration
 
-TODO
+Has two meanings:
+
+- Colloquially, a term referring to the many types of settings a player may use when generating a game. This includes trick settings, [Pickup Pool](#pickup-pool) settings, starting items and location, [Gameplay and Cosmetic Patches](#game-patches), and many other things.
+
+- In code, the Configuration class that each game must provide, derived from the BaseConfiguration class. This is where game-specific parts of the [Preset](#preset) are defined.
+
+## Connection
+
+Connects two [Nodes](#node) in the same area. Contains a set of [Requirements](#requirement) that define whether the [Generator](#generator) and [Solver](#solver) are allowed to cross it.
 
 ## Dangerous Action
 
-TODO
+Any action that the [Generator](#generator) might take which could potentially leave it stranded; a one-way action. Specifically, performing a Dangerous Action may leave the Generator unable to access [Pickups](#pickup) or other [Progression](#progression) which they would have been able to obtain before.
+
+An Action is not Dangerous if it is the only available Action which would allow them to progress, as in this scenario the Generator was unable to obtain other Checks even before they performed the Action. See also [Point of No Return](#point-of-no-return).
```

Comment:
Wrong. If only dangerous actions remain and the `logical_resource_action` was `never`, generation fails.

Reply:
Do you mind rewriting that bit? I was basing this almost completely off of TP's linked discord posts, your framework for this seems to be more implementation focused, which I am less knowledgeable of.

Reply:
I would just remove the second part, not relevant to know what a dangerous action is.

And the first part, it can be simplified to something like this:
> An action which, after being performed, prevents the access to nodes that were previously accessible.

Reply:
Alright, thanks. That's much more concise.